### DISCUSSION TITLE
Handle document update/delete in Meilisearch when product is being deleted/updated

### DISCUSSION THREAD

**Discussion by winston0410** [medusajs/medusa#4227] [Feature Requests]
Right now with `medusa-plugins-meilisearch`, if we deleted products in admin panel, those products will still show up in the search. And for new products we create in panel, they do not show up

My suggestion is
- we handle document creation in `product.create` event
- handle document update in `product.update` event
- handle document delete in `product.delete` event

**Comment by pevey** [comment]
If you have your medusa server open in dev mode in a console, and you update a product in the admin (example: add the word "test" to the product description), do you not see in the console that the product.updated event is called with (at least) 1 subscriber?  Is the word "test" not in the meilisearch hits after the update?  

If not, I would delete your data.ms file, update Medusa and the meilisearch plugin to the latest version, and try again.

**Reply by winston0410** [reply]
Are you saying that these 3 events are handled already, and I just missed them in the source of the meilisearch plugin?

**Reply by pevey** [reply]
Did you give it a try?  The subscribers are in the core.  https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/subscribers/product.ts

**Reply by winston0410** [reply]
Indeed, I have not read that part and I thought everything related was in that plugin, but I understand the design now and that make sense. Apology for causing the confusion and thank you so much for your answer

**Reply by pevey** [reply]
No apology needed.  I used the meilisearch plugin, so it's great to see people looking to contribute.  I just didn't want you to expend the effort unnecessarily.
